packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,225 @@
+# Bottom Sheet
+
+Bottom sheet is a container that presents additional information to a user in an overlay.
+
+This document outlines the API of `BottomSheet` component.
+
+<img src="./bottom-sheet-thumbnail.png" width="380" alt="" />
+
+## Design
+
+- [Figma - BottomSheet](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=26477%3A578228&t=1RMqjs99P33Udj0C-0)
+
+## API
+
+Sample usage:
+
+```jsx
+import { BottomSheet } from '@razorpay/blade';
+
+<BottomSheet open={boolean} snapPoints={[]}>
+  <BottomSheetHeader>
+    <BottomSheetHeaderLeading title="Payments Links" prefix={BladeIcon} />
+    <BottomSheetHeaderTrailing visual={Link | Action | Icon} />
+  </BottomSheetHeader>
+  <BottomSheetBody>Body Content</BottomSheetBody>
+  <BottomSheet>
+    <BottomSheetLeading title="Footer Title" prefix={icon} />
+    <BottomSheetTrailing
+      actions={{
+        primaryAction: { text: 'Know more', onClick: () => {} },
+        secondaryAction: { text: 'Read Docs', onClick: () => {} },
+      }}
+    />
+  </BottomSheet>
+</BottomSheet>;
+```
+
+**Design Constraints**
+
+- BottomSheetHeader:
+
+  - Leading:
+    - Title
+    - Prefix
+  - Trailing:
+    - empty
+    - icon
+    - link
+    - action
+
+- BottomSheetFooter:
+
+  - Leading
+    - title
+    - prefix
+  - Trailing
+    - Primary action button
+    - Secondary action button
+
+- The BottomSheetFooter will behave similar to Card's footer.
+- The Header, Footer can be individually omitted
+
+### `BottomSheet`
+
+We'll expose an `BottomSheeet` component with the following API:
+
+| Prop            | Type        | Default                 | Description                                                                                     | Required |
+| --------------- | ----------- | ----------------------- | ----------------------------------------------------------------------------------------------- | -------- |
+| open            | `boolean`   | `false`                 | toggles bottom sheet state content                                                              |          |
```

Comment:
Yes that `open` prop is a boolean that you pass `<BottomSheet open={true} />`

Do you mean to rename it to `isOpen`?

Reply:
+1

boolean values lets append them with `is`

Reply:
Right, and it should be required prop I guess?

Reply:
> Right, and it should be required prop I guess?

Can't make it required since the bottom sheet needs work as both "independent" & a "composed" component. 

when I say Independent i mean: 

```jsx
const [userPassedIsOpen, setIsOpen] = React.useState(false);

<BottomSheet isOpen={userPassedIsOpen}>
  <BottomSheetBody>some random stuff</BottomSheetBody>
</BottomSheet>

```

and also composable with Dropdown: 

```jsx
<Dropdown>
  <SelectInput />

  <BottomSheet> // <--- notice how the isOpen state is managed by the dropdown
    <BottomSheetBody>some random stuff</BottomSheetBody>
  </BottomSheet>
</Dropdown>
  
```